### PR TITLE
Blog App : Adding Forms and Like Button

### PR DESCRIPTION
**Pull request by ehmaddd**
The following features are added in this Pull Request :

- Created a method called current_user in ApplicationController to make current user data available to all controllers. It returns the first user from the database.

-------------------------------------------------------------------------------------------------------

Created forms to perform the following functions:
-  Create Post Form :  current_user creates the Post (using the method created in the ApplicationController).
-  Create Comment Form : current_user adds the Comment (using the method created in the ApplicationController).

--------------------------------------------------------------------------------------------------------

- Allowed users to add likes to Posts using "Like" button

--------------------------------------------------------------------------------------------------------

**GENERAL REQUIREMENTS :**

- There is no linter error.
- Used correct Gitflow.
- Documented the work in a professional way.
- Followed the list of best practices for Ruby

### CODE REVIEW

File: app/models/post.rb
Comment (IngDavidTM) [inline app/models/post.rb:1]: - [ ] While you did a fantastic job adding the forms 👍, as a project requirement and to create a new post there should be a new action called create which will help you to handle the implementation of the new post. 

Hint
```ruby

def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
```

File: app/controllers/posts_controller.rb
Comment (IngDavidTM) [inline app/controllers/posts_controller.rb:1]: - [x] While you did a fantastic job adding the forms 👍, as a project requirement and to create a new post there should be a new action called create which will help you to handle the implementation of the new post. 

Hint
```ruby

    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```

File: app/controllers/comments_controller.rb
Comment (IngDavidTM) [inline app/controllers/comments_controller.rb:1]: - [x] While you did a fantastic job adding the forms 👍, as a project requirement and to create a new comment there should be a new action called create which will help you to handle the implementation of the new comment. 

Hint
```ruby

def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
```

File: config/routes.rb
Comment (IngDavidTM) [inline config/routes.rb:1]: - [x] Kindly don’t forget to add `create` and `new` actions for posts 👍

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @ehmaddd  🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb
Comment (ehmaddd) [reply app/controllers/posts_controller.rb:1]: Changed as instructed

File: app/controllers/comments_controller.rb
Comment (ehmaddd) [reply app/controllers/comments_controller.rb:1]: Changed as desired

File: config/routes.rb
Comment (ehmaddd) [reply config/routes.rb:1]: Done as instructed

**Review by Davidosky007 — APPROVED** [review]
Hi @ehmaddd 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Forms created properly 👍 
* Well written PR 👍 
* No linter errors 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**